mitosheet/src/mito/components/endo/GridData.tsx

```diff
@@ -71,8 +70,15 @@ const GridData = (props: {
 
                             const isColumnCreated = reconIsColumnCreated(columnHeader, props.uiState.dataRecon, sheetData)
                             const isColumnModified = reconIsColumnModified(columnHeader, props.uiState.dataRecon, sheetData)
-                            const matchesSearch = checkMatchesSearch(cellData.toString(), props.uiState.currOpenSearch.searchValue);
 
+                            // Check if the cell is a search match. 
+                            const matchesSearch = !!props.uiState.currOpenSearch.matches.find((value) => {
```

Comment:
Oh nice! This allows us to reuse the logic, so we can be sure backend is the single source of truth